apps/qos-prioritizer/simulation/prepare_env.sh

```
@@ -0,0 +1,367 @@
+
+# Check if QoS service is already running
+if curl -s http://localhost:3005/health >/dev/null 2>&1; then
+    log_success "QoS service is already running on port 3005"
+else
+    log_info "Starting QoS prioritizer service..."
+    
+    # Kill any existing process on port 3005
+    kill_port 3005
+    
+    # Navigate to QoS service directory
+    cd "$PROJECT_ROOT"
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)